File: BOJ/#1929 소수 구하기/소수 구하기.py
M,N = map(int, input().split())

isPrime = [True for i in range(N+1)]
isPrime[1] = False

for i in range(4, N+1, 2): isPrime[i] = False
for i in range(3, N+1, 2):
    if isPrime[i]:
        for j in range(i*2, N+1, i): isPrime[j] = False

for i in range(M, N+1):
    if isPrime[i]: print(i)

File: BOJ/#1929 소수 구하기/소수 구하기.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int M,N; cin>>M>>N;

    // 에라토스테네스의 체
    vector<bool> isPrime(N+1, true);
    isPrime[1] = false;

    for(int i=4; i<=N; i+=2) isPrime[i] = false;
    for(int i=3; i<=N; i+=2) {
        if(isPrime[i])
            for(int j=i*2; j<=N; j+=i) isPrime[j] = false;
    }

    for(int i=M; i<=N; i++)
        if(isPrime[i]) printf("%d\n", i);
}
